CMakeLists.txt

```diff
@@ -12,11 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 cmake_minimum_required(VERSION 3.10)
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake" ${CMAKE_MODULE_PATH})
-
 # set the project name
 project(velox)
 
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake" ${CMAKE_MODULE_PATH})
+
+# Add all options below
+option(VELOX_ENABLE_S3 "Build S3 Connector" OFF)
```

Comment:
Cc: @mshang816 and @funrollloops 

we were discussing last week about start adding these options. This will be the first :)

Reply:
also, I'm guessing that the initial plan is to leave is disabled in circleCI, right?